### ISSUE TITLE
ADWIN reset method

### ISSUE THREAD

**Issue by gmartinon** [scikit-multiflow/scikit-multiflow#99]
### Expected behaviour

I would expect the `ADWIN.reset()` method to be equivalent to instantiate a fresh new ADWIN detector, in particular with a width, variance, total and number of detected changes all equal to zero.

### Actual behaviour

The actual implementation of the `reset()` method does not refresh the width nor the number of detected changes.

I understand that in ensemble learning methods (e.g. leverage_bagging), ADWIN is systematically dropped and rebuilt from scratch every time a new learner comes into play, which is ok since tree and detector are dropped together. 

But for pure detection purposes, shouldn't the `reset()` method be self-consistent and forget information from previous detection and previous data ?

---
[*Note:* For comments and usage-related questions please consider the [users group](https://groups.google.com/forum/#!forum/scikit-multiflow-users).

**Comment by jacobmontiel** [comment]
I agree that the current implementation of the `reset()` method differs from the expected behavior.

Here is a fix:

```python
def reset(self):
    self.__init__(delta=self.delta)
```

which requieres updating the initialization method:

```python
def __init__(self, delta=.002):
    ...
    super().reset()   # instead of self.reset()
```

This way we will set the object to its initial state (as expected).

**Comment by jacobmontiel** [comment]
Fixed in e2783e71f3a1d312529547fe4b399de96e270108
